outcomes/firefox_desktop/spotlight_engagement.toml

```diff
@@ -1,24 +1,41 @@
-friendly_name = "Spotlight Engagement - not experiment specific"
+### outcome properties
+friendly_name = "Spotlight Engagement - not experiment specific"  # TODO: is the not experiment specific still accurate?
 description = "Usage and engagement metrics for the Spotlight feature - for ALL engagement with Spotlight feature, not experiment specific "
 
+
+### parameters definition
+[parameters]
+
+[parameters.message_id]
+friendly_name = "Message ID associated with the experiment"
+description = "ID of the spotlight message associated with this experiment"
+default = "'AND DIV(7, 0)' -- Failed to specify a value for message_id"
+
+[parameters.branch]
+friendly_name = "Branch name associated with the experiment"
+description = "Branch name associated with this experiment's message id"
+default = "'AND DIV(11, 0)' -- Failed to specify a value for branch_id"
+
+
+### metrics definition
 [metrics.spotlight_impressions]
 friendly_name = "Spotlight Impressions"
 description = "How often users saw Spotlight during an analysis window"
-select_expression = "COUNTIF(event = 'IMPRESSION')"
+select_expression = "COUNTIF(event = 'IMPRESSION' AND message_id = '{{ parameters.message_id }}' AND e.branch = '{{ parameters.branch }}')"
 data_source = "spotlight"
 statistics = { bootstrap_mean = {} }
 
 [metrics.spotlight_clicks]
 friendly_name = "Spotlight Clicks"
 description = "How often users clicked Spotlight during an analysis window"
-select_expression = "COUNTIF(event = 'CLICK')"
+select_expression = "COUNTIF(event = 'CLICK' AND message_id = '{{ parameters.message_id }}' AND e.branch = '{{ parameters.branch }}')"
 data_source = "spotlight"
 statistics = { bootstrap_mean = {} }
 
 [metrics.spotlight_dismisses]
 friendly_name = "Spotlight Dismisses"
 description = "How often users dismissed Spotlight during an analysis window"
-select_expression = "COUNTIF(event = 'DISMISS')"
+select_expression = "COUNTIF(event = 'DISMISS' AND message_id = '{{ parameters.message_id }}' AND e.branch = '{{ parameters.branch }}')"
```

Comment:
```suggestion
select_expression = "COUNTIF(event = 'DISMISS' AND message_id = '{{ parameters.message_id }}')"
```

Reply:
I thought we needed to specify both? 

Anyways, the plan if to first make a change that supports config for a single message_id. I opened a PR which I think is adding this ability. After we're happy with that change, I'd like to merge and then make a separate PR adding support for multiple message_ids and branches.